docs/source/dataset_vs_iterable_dataset.mdx

```diff
@@ -0,0 +1,217 @@
+# Choose between Dataset and IterableDataset
+
+## Downloading and streaming
+
+When you have a regular "map-style" [`Dataset`], you can access it using `my_dataset[0]`: we have what we call "random access" to the rows.
```

Comment:
What is meant by a “map-style” `Dataset`? If I understand correctly, this is just a regular `Dataset`. So it might be easier for users to understand if we don’t use this specific term and just use `Dataset` or if we define what we mean by “map-style” (unless this is commonly known jargon, in which case ignore this haha).

Reply:
Yes it refers to datasets with random access, i.e. that allows you to do `my_dataset[0]`. I'll define it properly